near-sdk/Cargo.toml

```diff
@@ -24,7 +24,7 @@ serde_json = "1"
 near-sdk-macros = { path = "../near-sdk-macros", version = "~4.1.1" }
 near-sys = { path = "../near-sys", version = "0.2" }
 base64 = "0.13"
-borsh = { version = "0.9", features = ["const-generics"] }
+borsh = { version = "1.0.0-alpha.3", features = ["derive", "schema"] }
```

Comment:
Is `schema` only needed by `near-abi`? Sounds like this feature only needs to be enabled when `abi` is enabled.

Reply:
Nope, there're direct imports outside of `abi` feature's scope:

```bash
cargo check -p near-sdk --no-default-features
```

```bash
 1  error[E0432]: unresolved import `borsh::BorshSchema`
  --> near-sdk/src/promise.rs:1:5
 2  error[E0432]: unresolved imports `borsh::schema`, `borsh::BorshSchema`
  --> near-sdk/src/private/metadata.rs:1:13
 3  error[E0432]: unresolved import `borsh::BorshSchema`
  --> near-sdk/src/types/account_id.rs:1:31
 4  error[E0432]: unresolved import `borsh::BorshSchema`
  --> near-sdk/src/types/gas.rs:1:31
 5  error: cannot determine resolution for the derive macro `BorshSchema`
  --> near-sdk/src/private/metadata.rs:6:44
 6  error: cannot determine resolution for the derive macro `BorshSchema`
   --> near-sdk/src/private/metadata.rs:21:44
 7  error: cannot determine resolution for the derive macro `BorshSchema`
   --> near-sdk/src/types/account_id.rs:37:84
 8  error: cannot determine resolution for the derive macro `BorshSchema`
   --> near-sdk/src/types/gas.rs:18:5
 9  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:232:43
10  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:232:71
11  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:237:32
12  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:577:43
13  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:577:71
14  error[E0433]: failed to resolve: could not find `schema` in `borsh`
    --> near-sdk/src/promise.rs:582:32

```
There is, e.g., 
```rust
impl BorshSchema for Promise {

```

but 
```rust
#[cfg(feature = "abi")]
impl schemars::JsonSchema for Promise {

```

Reply:
Some extra-redundant cross-referencing: [pr](https://github.com/near/borsh-rs/pull/210)

Reply:
@dj8yfo Please, feature-gate uses of BorshSchema to be part of `abi` feature 🙏